charts/kube-ovn-v2/values.yaml

```
@@ -0,0 +1,719 @@
+    # @section -- Network parameters of the CNI
+    # @default -- "{}"
+    cidr:
+      # -- IPv4 CIDR.
+      # @section -- Network parameters of the CNI
+      v4: "10.96.0.0/12"
+      # -- IPv6 CIDR.
+      # @section -- Network parameters of the CNI
+      v6: "fd00:10:96::/112"
+  # -- Configuration of the "join" subnet, used by the nodes to contact (join) the pods in the default subnet.
+  # If .networking.stack is set to IPv4, only the .v4 key is used.
+  # If .networking.stack is set to IPv6, only the .v6 key is used.
+  # If .networking.stack is set to Dual, both keys are used.
+  # @section -- Network parameters of the CNI
+  # @default -- "{}"
+  join:
+    # -- Name of the join subnet once it gets generated in the cluster.
+    # @section -- Network parameters of the CNI
+    subnetName: "join"
+    # -- CIDR used by the join subnet.
+    # @section -- Network parameters of the CNI
+    # @default -- "{}"
+    cidr:
+      # -- IPv4 CIDR.
+      # @section -- Network parameters of the CNI
+      v4: "100.64.0.0/16"
+      # -- IPv6 CIDR.
+      # @section -- Network parameters of the CNI
+      v6: "fd00:100:64::/112"
+
+  # -- Name of the default VPC once it is generated in the cluster.
+  # Pods in the default subnet live in this VPC.
+  # @section -- Network parameters of the CNI
+  defaultVpcName: "ovn-cluster"
+  # -- Deploy the CNI with SSL encryption in between components.
+  # @section -- Network parameters of the CNI
+  enableSsl: false
+  # -- Network type can be "geneve" or "vlan".
+  # @section -- Network parameters of the CNI
+  networkType: geneve
+  # -- Tunnel type can be "geneve", "vxlan" or "stt".
+  # @section -- Network parameters of the CNI
+  tunnelType: geneve
+  # -- IPs to exclude from IPAM in the default subnet.
+  # @section -- Network parameters of the CNI
+  excludeIps: ""
+  # -- NIC type used on pods to connect them to the CNI.
+  # @section -- Network parameters of the CNI
+  podNicType: "veth-pair"
+  # -- Enable EIP and SNAT.
+  # @section -- Network parameters of the CNI
+  enableEipSnat: true
+  # -- Comma-separated string of NodeLocal DNS IP addresses.
+  # @section -- Network parameters of the CNI
+  nodeLocalDnsIp: ""
+  # -- Enable listening on the metrics endpoint for the CNI daemons.
+  # @section -- Network parameters of the CNI
+  enableMetrics: true
+
+  # -- Configuration if we're running on top of a VLAN.
+  # @section -- Network parameters of the CNI
+  vlan:
+    providerName: "provider"
+    interfaceName: ""
+    name: "ovn-vlan"
+    id: "100"
+
+  # -- ""
+  # @section -- Network parameters of the CNI
+  exchangeLinkName: false
+  # -- ""
+  # @section -- Network parameters of the CNI
+  enableEcmp: false
+  # -- ""
+  # @section -- Network parameters of the CNI
+  enableCompact: false
+
+# -- Features of Kube-OVN we wish to enable/disable.
+# @section -- Opt-in/out Features
+features:
+  ENABLE_LB: true
+  ENABLE_NP: true
+  ENABLE_EXTERNAL_VPC: true
```

```suggestion
  EnableLoadBalacner: true
  EnableNetworkPolicy: true
  EnableExternalVPC: true
```

Let's use k8s feature-gate style names.

---

I'm gonna work on this. I'll still use the standard camel case syntax to respect Helm standards.
There's a lot of work to do on the old "func" dict, I didn't touch it too much because I'm not sure what is still relevant or not.

---

@zhangzujian https://github.com/kubeovn/kube-ovn/pull/4437/commits/34574c575a85eae5b51c8653c457e8892692b637

this commit addresses some of the feature flags, I left some of them because I have no idea what they do. If you want to add some more comments to each flag to make the values more bearable to new users, feel free to give me feedback